src/ruby/ext/grpc/rb_call.c

```diff
@@ -179,6 +179,36 @@ static VALUE grpc_rb_call_cancel(VALUE self) {
   return Qnil;
 }
 
+/* Not a part of the surface API.
+ * This is meant for internal usage by the "write thread" of grpc-ruby
+ * client-side bidi calls. It provides a way for the background write-thread
+ * to propogate failures to the main read-thread and give the user an error
+ * message. */
+static VALUE grpc_rb_call_cancel_with_bidi_write_thread_error_message(
```

Comment:
That description is a read only string from grpc_call_cancel_with_status's point of view. That is why it is duplicated, so the error can take ownership of its life. Otherwise it's just printed. So surfacing grpc_call_cancel_with_status should be safe I think? You can pass it any char*, and it won't modify or deallocate it